plugin/plugin.go

```diff
@@ -0,0 +1,86 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+	"github.com/hashicorp/go-plugin"
+	gp "github.com/hashicorp/go-plugin"
+	grpc "google.golang.org/grpc"
+)
+
+// HandshakeConfig is a shared config that can be used regardless of wrapper, to
+// avoid having to know type-specific things about each plugin
+var HandshakeConfig = gp.HandshakeConfig{
+	MagicCookieKey:   "HASHICORP_GKW_PLUGIN",
+	MagicCookieValue: "wrapper",
+}
+
+// wrapper embeds Plugin and is used as the top-level
+type wrapper struct {
+	gp.Plugin
```

Comment:
I don't see this being set anywhere, am i missing it or is it coming in a follow on PR?

Reply:
Oh, or should this be

```suggestion
	// Embeding this will disable the netRPC protocol
	plugin.NetRPCUnsupportedPlugin
```